src/proto/single/ent.rs

```diff
@@ -0,0 +1,130 @@
+use chrono::{DateTime, Utc};
+
+use crate::JobBuilder;
+
+impl JobBuilder {
+    /// When Faktory should expire this job.
+    ///
+    /// Faktory Enterprise allows for expiring jobs. This is setter for `expires_at`
+    /// field in the job's custom data.
+    /// ```
+    /// use faktory::JobBuilder;
+    /// use chrono::{Duration, Utc};
+    ///
+    /// let _job = JobBuilder::new("order")
+    ///     .args(vec!["ISBN-13:9781718501850"])
+    ///     .expires_at(Utc::now() + Duration::hours(1))
+    ///     .build();
+    /// ```
+    pub fn expires_at(&mut self, dt: DateTime<Utc>) -> &mut Self {
+        self.add_to_custom_data(
+            "expires_at".into(),
+            dt.to_rfc3339_opts(chrono::SecondsFormat::Nanos, true),
+        )
+    }
+
+    /// In what period of time from now (UTC) the Faktory should expire this job.
+    ///
+    /// Use this setter when you are unwilling to populate the `expires_at` field in custom
+    /// options with some exact date and time, e.g.:
+    /// ```
+    /// use faktory::JobBuilder;
+    /// use chrono::Duration;
+    ///
+    /// let _job = JobBuilder::new("order")
+    ///     .args(vec!["ISBN-13:9781718501850"])
+    ///     .expires_in(Duration::weeks(1))
+    ///     .build();
+    /// ```
+    pub fn expires_in(&mut self, ttl: chrono::Duration) -> &mut Self {
+        self.expires_at(Utc::now() + ttl)
+    }
+
+    /// How long the Faktory will not accept duplicates of this job.
+    ///
+    /// The job will be considered unique for kind-args-queue combination.
+    /// The uniqueness is best-effort, rather than a guarantee. Check out
+    /// the Enterprise Faktory [docs](https://github.com/contribsys/faktory/wiki/Ent-Unique-Jobs)
+    /// for details on how scheduling, retries and other features live together with `unique_for`.
+    pub fn unique_for(&mut self, secs: usize) -> &mut Self {
```

Comment:
I think commonly `u64` is used for seconds in this setting (e.g., `std::time::Duration::as_secs`).

Reply:
This is to keep it consistent with`reserve_for` on the `Job` struct, which is an option of usize. 

Also - I've  just checked out of  curiosity - they use _uint_ in the Go [client](https://github.com/contribsys/faktory/blob/main/client/job.go#L103).

As for duration is seconds generally, in `chrono` the [`seconds`](https://docs.rs/chrono/latest/chrono/struct.Duration.html#method.seconds) method will also expect a 64bit integer, but  a _singed_ one, allowing for negative duration.